db/crud.go

```diff
@@ -797,9 +791,9 @@ func (db *Database) updateAndReturnDoc(
 			doc.syncData.TombstonedAt = 0
 		}
 
-		if doc.CurrentRev != prevCurrentRev && prevCurrentRev != "" && doc.body != nil {
+		if doc.CurrentRev != prevCurrentRev && prevCurrentRev != "" && doc.Body() != nil {
 			// Store the doc's previous body into the revision tree:
-			bodyJSON, _ := json.Marshal(doc.body)
+			bodyJSON, _ := doc.MarshalBody()
```

Comment:
Looks like the error is being ignored here.  I think it should at least log a warning.